### PR TITLE
refactor `actions/index.js`

### LINKED ISSUE #754: reorganize `static/js/actions/index.js`
In `actions/ui.js` we've been using the following style for exporting an action type and it's corresponding action creator:

```js
export const MY_ACTION_TYPE = 'MY_ACTION_TYPE';
export const myActionType = (arg: type): Action => (
  { type: MY_ACTION_TYPE, payload: arg }
);
```

So the declaration of the action type and it's corresponding action creator are right next to each other. I think this is nice and readable. In `actions/index.js` currently we declare all of the action types at the top in a big block and then all the action creators down below. I'd like to reorganize `actions/index.js` along the lines that the `ui` actions are currently organized.

While I'm at it I'm going to add type annotations.

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

this closes #754 

#### What's this PR do?

Small refactor of `static/js/actions/index.js`. Basically just moving some code around and adding type annotations.

#### Where should the reviewer start?

Read the code!

#### How should this be manually tested?

Functionality should be totally the same, so probably click around and make sure all the profile and dashboard related functionality is working correctly.

### CODE REVIEW

File: static/js/actions/index.js [hunk 1]
```diff
@@ -79,12 +82,14 @@ export const saveProfile = (username, profile) => {
...
+
+export const UPDATE_PROFILE_VALIDATION = 'UPDATE_PROFILE_VALIDATION';
+export const updateProfileValidation = (username: string, errors: ValidationErrors): Action => ({
   type: UPDATE_PROFILE_VALIDATION,
   payload: { errors, username }
 });
 
-export function fetchUserProfile(username) {
+export function fetchUserProfile(username: string): Dispatcher {
   return dispatch => {
```
Comment (gsidebo) [inline static/js/actions/index.js]: does it make sense to add the same type checking here/`saveProfile`/etc as you did in `fetchDashboard` below? (referring to `return (dispatch: Dispatch) => {`)

Comment (aliceriot) [inline static/js/actions/index.js]: yes! good catch

**Comment by aliceriot** [comment]
@gsidebo I added that annotation (for `dispatch`)

**Comment by gsidebo** [comment]
looks great! 👍

**Pull request merged**